src/components/views/beacon/RoomCallBanner.tsx

```diff
@@ -0,0 +1,131 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useCallback } from "react";
+import { MatrixEvent, Room } from "matrix-js-sdk/src/matrix";
+
+import { _t } from "../../../languageHandler";
+import AccessibleButton, { ButtonEvent } from "../elements/AccessibleButton";
+import dispatcher, { defaultDispatcher } from "../../../dispatcher/dispatcher";
+import { ViewRoomPayload } from "../../../dispatcher/payloads/ViewRoomPayload";
+import { Action } from "../../../dispatcher/actions";
+import { Call, ConnectionState, ElementCall } from "../../../models/Call";
+import { useCall } from "../../../hooks/useCall";
+import { RoomViewStore } from "../../../stores/RoomViewStore";
+import { useEventEmitterState } from "../../../hooks/useEventEmitter";
+import {
+    OwnBeaconStore,
+    OwnBeaconStoreEvent,
+} from "../../../stores/OwnBeaconStore";
+import { CallDurationFromEvent } from "../voip/CallDuration";
+
+interface RoomCallBannerProps {
+    roomId: Room["roomId"];
+    call: Call;
+}
+
+const RoomCallBannerInner: React.FC<RoomCallBannerProps> = ({
+    roomId,
+    call,
+}) => {
+    const callEvent: MatrixEvent | null = (call as ElementCall)?.groupCall;
+
+    const connect = useCallback(
+        (ev: ButtonEvent) => {
+            ev.preventDefault();
+            defaultDispatcher.dispatch<ViewRoomPayload>({
+                action: Action.ViewRoom,
+                room_id: roomId,
+                view_call: true,
+                metricsTrigger: undefined,
+            });
+        },
+        [roomId],
+    );
+
+    const onClick = useCallback(() => {
+        dispatcher.dispatch<ViewRoomPayload>({
+            action: Action.ViewRoom,
+            room_id: roomId,
+            metricsTrigger: undefined,
+            event_id: callEvent.getId(),
+            scroll_into_view: true,
+            highlighted: true,
+        });
+    }, [callEvent, roomId]);
+
+    return (
+        <div
+            className="mx_RoomCallBanner"
+            onClick={onClick}
+        >
+            <div className="mx_RoomCallBanner_text">
+                <span className="mx_RoomCallBanner_label">{ _t("Video call") }</span>
+                <CallDurationFromEvent mxEvent={callEvent} />
+            </div>
+
+            <AccessibleButton
+                onClick={connect}
+                kind="primary"
+                element="button"
+                disabled={false}
+            >
+                { _t("Join") }
+            </AccessibleButton>
+        </div>
+    );
+};
+
+interface Props {
+    roomId: Room["roomId"];
+}
+
+const RoomCallBanner: React.FC<Props> = ({ roomId }) => {
+    const call = useCall(roomId);
+
+    // this section is to check if we have a live location share. If so, we dont show the call banner
```

Comment:
If the comment could mention why, it could be helpful - I'm assuming because they just occupy the same space, in which case this i obviously a bit nonideal & we probably want something a bit more like the toast manager that decides which one takes precedence (but no need to block on that part).

Reply:
yes, carry also proposed a better solution than what it currently is. (this makes testing easier though (I think) since it is all inside one component)